crates/bevy_hierarchy/src/child_builder.rs

```diff
@@ -279,6 +279,50 @@ pub struct ChildBuilder<'a> {
     add_children: AddChildren,
 }
 
+/// Provides helper methods for getting the [`Commands`] struct used to create the provided child
+/// builder. Useful for accessing an extension trait or method on commands from within the child
+/// builder
+///
+/// # Example
+///
+/// This example creates three entities and then triggers an event for an observer
+///
+/// ```
+/// # use bevy_ecs::bundle::Bundle;
+/// # use bevy_ecs::Trigger;
+/// # use bevy_ecs::Event;
+/// # use bevy_ecs::system::Commands;
+/// # use bevy_hierarchy::{ChildBuild, BuildChildren};
+/// # #[derive(Event)]
+/// # struct AllChildrenCreated;
+/// # #[derive(Bundle)]
+/// # struct MyBundle {}
+/// # #[derive(Bundle)]
+/// # struct MyChildBundle {}
+/// #
+/// # fn test(mut commands: Commands) {
+/// // See source for the full imports so that you can run this example
+/// commands.spawn(MyBundle {}).with_children(|child_builder| {
+///     child_builder.spawn(MyChildBundle {});
+///     child_builder.spawn(MyChildBundle {});
+///     child_builder.commands_mut().trigger(AllChildrenCreated);
+/// });
+/// # }
+/// ```
+impl<'a> ChildBuilder<'a> {
+    /// Returns the [`Commands`] used to create this [`ChildBuilder`] object, usually from calling
+    /// [`Commands::spawn`], spawning an entity. The lifetime used in `commands_mut` and `commands`
+    /// is longer than `Self`
+    pub fn commands_mut(&mut self) -> &mut Commands<'a, 'a> {
+        &mut self.commands
+    }
+
+    /// Same as above, except non-mutable
+    pub fn commands(&self) -> &Commands<'a, 'a> {
```

Comment:
I'm not sure there's a use-case for an immutable reference to a `Commands`. I think something like this might work?
```rust
pub fn commands(&mut self) -> Commands { self.commands.reborrow() }
```

Reply:
Agreed, I would just have the one method that returns a mutable commands.

Is there a reason we're not simply returning `self.world.commands()`?

Reply:
Since this is a method on `ChildBuilder`, there's no `self.world`.